File: src/main/java/com/learnershi/rclasssocket/repository/BadgeRepository.java
package com.learnershi.rclasssocket.repository;

import com.learnershi.rclasssocket.entity.Badge;
import com.learnershi.rclasssocket.entity.enums.BadgeType;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface BadgeRepository extends ReactiveMongoRepository<Badge, String> {
    Flux<Badge> findAllByClassRoomId(String classRoomId);
    Flux<Badge> findAllByUserSeq(String userSeq);
    Mono<Badge> findByClassRoomIdAndUserSeqAndBadgeType(String classRoomId, String userSeq, BadgeType badgeType);
}

File: src/main/java/com/learnershi/rclasssocket/repository/BadgeRepository.kt
package com.learnershi.rclasssocket.repository

import com.learnershi.rclasssocket.entity.Badge
import com.learnershi.rclasssocket.entity.enums.BadgeType
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface BadgeRepository : ReactiveMongoRepository<Badge?, String?> {
    fun findAllByClassRoomId(classRoomId: String?): Flux<Badge?>
    fun findAllByUserSeq(userSeq: String?): Flux<Badge?>
    fun findByClassRoomIdAndUserSeqAndBadgeType(
        classRoomId: String?,
        userSeq: String?,
        badgeType: BadgeType?
    ): Mono<Badge?>?
}